File: Interview Preparation Kit/Warm-up Challenges/Counting Valleys.py
def countingValleys(n, s):
    if n > 10 ** 6 or n < 2:
        return None
    valley = 0
    highlv = 0
    string_arr = list(s)
    for val in string_arr:
        if not (val == 'U' or val == 'D'):
            return None
        if val == 'U':
            highlv += 1
        elif val == 'D':
            highlv -= 1
            if highlv == -1:
                valley += 1
    return valley
print(countingValleys(15,'UDDUUUUDDDDUUUD'))

File: Interview Preparation Kit/Warm-up Challenges/Counting Valleys.js
'use strict';

function countingValleys(n, s) {
    if ((n < 2) || (n > Math.pow(10, 6))) {
        return;
    }
    let highlv = 0;
    let steparr = s.split('');
    let valley = 0;
    for (let val of steparr) {
        if ((val != 'D') && (val != 'U')) {
            return;
        }
        if (val == 'D') {
            highlv--;
            if ((highlv == -1) && (val == 'D')) {
                valley++;
            };
            continue;
        }
        if (val == 'U') {
            highlv++;
        };
    };
    return valley;
}

console.log(countingValleys(8, 'UDDDUDUU'))